core/isolate.rs

```diff
@@ -68,15 +68,21 @@ pub enum StartupData<'a> {
   None,
 }
 
-/// Defines the how Deno.core.dispatch() acts.
-pub trait Dispatch {
+#[derive(Default)]
+pub struct Config {
+  dispatch: Option<Arc<Fn(&[u8], deno_buf) -> (bool, Box<Op>) + Send + Sync>>,
+}
+
+impl Config {
+  /// Defines the how Deno.core.dispatch() acts.
   /// Called whenever Deno.core.dispatch() is called in JavaScript. zero_copy_buf
   /// corresponds to the second argument of Deno.core.dispatch().
-  fn dispatch(
-    &mut self,
-    control: &[u8],
-    zero_copy_buf: deno_buf,
-  ) -> (bool, Box<Op>);
+  pub fn dispatch<F>(&mut self, f: F)
+  where
+    F: Fn(&[u8], deno_buf) -> (bool, Box<Op>) + Send + Sync + 'static,
```

Comment:
hmm why does F need Send and Sync ?

Reply:
F is called on random threads, so it needs to be thread safe